storey/aggregations.py

```diff
@@ -188,15 +185,13 @@ class Persist(Flow):
     def __init__(self, table):
         super().__init__()
         self._table = table
+        self._closeables = [table]
 
-    async def _do(self, event):
-        if event is _termination_obj:
-            await self._table.close()
-            return await self._do_downstream(_termination_obj)
-        else:
-            # todo: persist keys in parallel
-            await self._table._persist_key(event.key)
-            await self._do_downstream(event)
+    async def _process_event(self, event):
+        return await self._table.persist_key(event.key)
+
+    async def _handle_completed(self, event, response):
+        await self._do_downstream(event)
```

Comment:
`response` is never used?

Reply:
I guess the idea is that is might be in some other impl?

Reply:
don't think there is any response for the `_do_downstream` other than on termination. which is handled elsewhere